forte/data/readers/sem_eval_task8_reader.py

```diff
@@ -0,0 +1,134 @@
+        and one RelationLink,
+        link = RelationLink(parent=Peopel, child=downtown)
+        link.rel_type = Entity-Destination
+        into the DataPack
+    """
+    def _cache_key_function(self, file_path: str) -> str:
+        return os.path.basename(file_path)
+
+    def _collect(self, *args, **kwargs) -> Iterator[Any]:
+        # pylint: disable = unused-argument
+        # args[0] should be the floder where
+        # SemEval Task8 dataset is stored.
+        # Files ended with .txt are exptected here.
+        sem_file_dir = args[0]
+        return dataset_path_iterator(sem_file_dir,
+            self.configs.sem_eval_task8_file_extension)
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        pack = self.new_pack()
+        fp = open(file_path, 'r', encoding='utf8')
+        txt = ""
+        offset = 0
+
+        while True:
+            sent_line = fp.readline()
+            if not sent_line:
+                break
+
+            if len(sent_line.split()) == 0:
+                continue
+
+            relation_line = fp.readline()
+            # command line is not used
+            _ = fp.readline()
+
+            sent_line = sent_line[sent_line.find('"') + 1:
+                                sent_line.rfind('"')]
+            index1 = sent_line.find("<e1>")
+            index2 = sent_line.find("<e2>")
+            e1 = sent_line[index1:sent_line.find("</e1>") + 5]
+            e2 = sent_line[index2:sent_line.find("</e2>") + 5]
+            sent_line = sent_line.replace(e1, e1[4:-5])
+            sent_line = sent_line.replace(e2, e2[4:-5])
+            e1 = e1[4:-5]
+            e2 = e2[4:-5]
+
+            if index1 < index2:
+                diff1 = 0
+                diff2 = 9
+            else:
+                diff1 = 9
+                diff2 = 0
+            index1 += offset - diff1
+            index2 += offset - diff2
+
+            Sentence(pack, offset, offset + len(sent_line))
+            entry1 = EntityMention(pack, index1, index1 + len(e1))
+            entry2 = EntityMention(pack, index2, index2 + len(e2))
+            offset += len(sent_line) + 1
+            txt += sent_line + " "
+
+            pair = relation_line[relation_line.find("(") + 1:
+                        relation_line.find(")")]
+
+            if "," in pair:
+                parent, _ = pair.split(",")
+                if parent == "e1":
+                    relation = RelationLink(pack, entry1, entry2)
+                else:
+                    relation = RelationLink(pack, entry2, entry1)
+                relation.rel_type = relation_line[:relation_line.find("(")]
+            else:
+                # for "Other" relation, just set parent as e1
+                # set child as e2
+                relation = RelationLink(pack, entry1, entry2)
+                relation.rel_type = relation_line.strip()
+
+        pack.set_text(txt, replace_func=self.text_replace_operation)
+        pack.pack_name = file_path
```

Comment:
`file_path` is not very ideal as the name. It will include some system dependent paths. Can you extract the base name?

Reply:
Hi. Thank you for the comment. Update now.